### ISSUE TITLE
Unify treatment of mean molecular mass and temperature

### ISSUE THREAD

**Issue by forrestglines** [parthenon-hpc-lab/athenapk#45]
As mentioned in #2 about `src/cluster/agn_triggering.cpp`, it would be best to unify treatment of He mass fraction, mean molecular mass, and computing temperature.

**Comment by pgrete** [comment]
Tightly couple (if not duplicate of #22) and I'm definitely in favor of this.
We should probably make this happen once the cluster PR is merged as it should be fairly straightforward.

**Comment by pgrete** [comment]
This update should also include an input parameter for the Coulomb logarithm (which, e.g., enters the calculation of the thermal conductivity).
